go/src/koding/kites/kloud/stack/provider/builder.go

```diff
@@ -217,6 +217,20 @@ func (b *Builder) BuildStackTemplate(templateID string) error {
 	return nil
 }
 
+// Authorize verifies whether user is allowed to access b.StackTemplate.
+//
+// Prior calling to this method it is required to build the stack
+// template first.
+//
+// If the stack template was not built, the method is a nop.
+func (b *Builder) Authorize(username string) error {
+	if b.StackTemplate == nil {
```

Comment:
Even if the data is corrupted, the model itself is never nil, in the worst case it can be zero value.

Either way should we add checks to all functions in modelhelper package that take pointer arguments (models)?